test/dataloader2/test_mprs.py

```diff
@@ -276,10 +276,32 @@ def test_reading_service_limit(self, dp, n_workers, worker_prefetch_cnt, main_pr
     # cumulative_res.extend(res)
     # self.assertEqual(list(range(n_elements)), sorted(cumulative_res))
 
+    @mp_ctx_parametrize
+    @dp_parametrize
+    @parametrize("n_workers,worker_prefetch_cnt,main_prefetch_cnt", [(2, 1, 1), (4, 1, 0), (4, 0, 0)])
+    def test_reading_service_snapshot(self, ctx, dp, n_workers, worker_prefetch_cnt, main_prefetch_cnt) -> None:
+        # Functional Test: Confirms that `snapshot` does capture the state of the underlying DataPipes properly
+        rs = MultiProcessingReadingService(
+            num_workers=n_workers,
+            worker_prefetch_cnt=worker_prefetch_cnt,
+            main_prefetch_cnt=main_prefetch_cnt,
+            multiprocessing_context=ctx,
+        )
+        dl: DataLoader2 = DataLoader2(dp, reading_service=rs)
+        res = []
+        for i, x in enumerate(dl):
+            res.append(x)
+            if i in {2}:
+                snapshot = dl.reading_service.snapshot()
```

Comment:
As you mention in the description, I could indeed see it making sense to test that the state in the snapshot is as we actually expect. At the very least for a prefetch=0 case.